auth/providers/azure/azure.go

```diff
@@ -226,10 +229,16 @@ func (s Authenticator) Check(ctx context.Context, token string) (*authv1.UserInf
 		if err := s.graphClient.RefreshToken(ctx, token); err != nil {
 			return nil, err
 		}
-		resp.Groups, err = s.graphClient.GetGroups(ctx, resp.Username)
+		if s.Options.AuthMode == ARCAuthMode {
+			resp.Groups, err = s.graphClient.GetMemberGroupsUsingARCOboService(ctx, s.Options.TenantID, s.Options.ResourceId, s.Options.AzureRegion, token)
+		} else {
```

Comment:
can we move this info while creating client and have same signature for `GetGroups`?

Reply:
done